gslib/tests/test_storage_url.py

```diff
@@ -135,3 +136,34 @@ def test_urls_are_mix_of_objects_and_buckets_is_true_for_a_mix(self):
   def test_urls_are_mix_of_objects_and_buckets_is_null_for_invalid(self):
     urls = list(map(storage_url.StorageUrlFromString, ['gs://b', 'f:o@o:o']))
     self.assertIsNone(storage_url.UrlsAreMixOfBucketsAndObjects(urls))
+
+  def test_urls_raise_error_if_bucket_followed_by_object(self):
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b1', 'gs://b/o']))
+    with self.assertRaisesRegex(
+            CommandException,
+            'Cannot operate on a mix of buckets and objects.'):
+      storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=False)
+
+  def test_urls_raise_error_if_object_followed_by_bucket(self):
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b/o', 'gs://b']))
+    with self.assertRaisesRegex(
+            CommandException,
+            'Cannot operate on a mix of buckets and objects.'):
+      storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=False)
+
+  def test_accepts_mix_of_objects_and_buckets_if_recursion_requested(self):
+    # No error raised
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b1', 'gs://b/o']))
+    storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=True)
+
+  def test_not_raising_error_if_multiple_objects_without_recursion(self):
+    urls = list(map(storage_url.StorageUrlFromString, ['gs://b/o', 'gs://b/p']))
+    self.assertFalse(storage_url.RaiseErrorIfUrlsAreMixOfBucketsAndObjects(urls, recursion_requested=False))
```

Comment:
Don't need to `assertFalse` on these. We're implicitly testing an error wasn't raised.

Reply:
Looks like you missed some